In [ ]:
from agent_team import AgentTeam
from database import CervicalCancerDB
from rag import RAG

In [3]:
# setup DB
db = CervicalCancerDB()
print("all patient names in DB:")
print(db.get_all_patient_names())

# setup RAG based on DB
rag = RAG(db)

# setup autoplanning team
team = AgentTeam(db, rag)

# setup planning task
init_msg = team.get_initmsg(patient_name="ZXY", top_n=2)

# run auto planning
team.run_auto_planning(init_msg) 

all patient names in DB:
['A', 'B', 'C', 'D', 'E', 'F', 'G', 'H', 'I', 'ZXY']
Reference Plans based on OARs' Overlap Percentages Similarity:
    Patient-Name    Rectum   Bladder  Bowel_bag  Femoral_head_L  \
0  Query Patient  0.126497  0.286670   0.041891             0.0   
1              I  0.258065  0.193435   0.032911             0.0   
2              H  0.361328  0.403633   0.025002             0.0   

   Femoral_head_R  Similarity Score  
0             0.0               NaN  
1             0.0          0.896024  
2             0.0          0.834110  
task_initializer (to chat_manager):


# Planning Task: Optimize radiotherapy plan for cervical cancer patient ZXY.

## Patient ZXY Profile:

### Anatomy

- Structures and Volumes:

| ROIs                       | Volume (cm³)              |
|----------------------------|---------------------------|
| PTV                        | 1078.50  |
| Bladder                    | 512.74  |
| Rectum                     | 119.44  |
| Bowel Bag    

The dosimetric outcomes of the optimized plan generated by Pinnacle TPS:  PTV: D95 is 5063.19 cGy; Max Dose is 5758.2 cGy; Min Dose is 4201 cGy; Mean Dose is 5301.3cGy.  Bladder: Max Dose is 5648.7 cGy; Mean Dose is 3759.1 cGy; V50 is 29%; V40 is 48%; V30 is 68%; V20 is 85%; V10 is 99%.  Rectum: Max Dose is 5534.9 cGy; Mean Dose is 3989.3 cGy; V50 is 42%; V40 is 59%; V30 is 70%; V20 is 85%; V10 is 100%.  BowelBag: Max Dose is 5524.2 cGy; Mean Dose is 1346.5 cGy; V50 is 5%; V40 is 7%; V30 is 11%; V20 is 23%; V10 is 48%.  FemoralHead_L: Max Dose is 4585 cGy; Mean Dose is 1712.1 cGy; V50 is 0%; V40 is 2%; V30 is 8%; V20 is 29%; V10 is 83%.  FemoralHead_R: Max Dose is 4481.6 cGy; Mean Dose is 1751.7 cGy; V50 is 0%; V40 is 0.39%; V30 is 7%; V20 is 32%; V10 is 86%